blocks/image-switcher/image-switcher.js

```diff
@@ -0,0 +1,89 @@
+import { createOptimizedPicture } from '../../scripts/aem.js';
+import {
+  ul,
+  li,
+  div,
+  p,
+  domEl,
+} from '../../scripts/dom-helpers.js';
+
+export default function decorate(block) {
+  block.querySelectorAll('img').forEach((image) => image
```

Comment:
Is this a performance optimization? Have you measured its effect?

Reply:
I haven't measured the effect, but it is a performance optimization, used in almost all blocks if I'm not mistaken